R/gh_response.R

```diff
@@ -44,6 +44,10 @@ gh_error <- function(response, call = sys.call(-1)) {
   res <- content(response)
   status <- status_code(response)
 
+  if (!length(res)) {
+    res <- list(message = "Empty Body\n")
+  }
+
```

Comment:
Small drive-by observation:I may be taking this out of context.

Should this really be manufacturing a body / a string, where one did not previously exist?

In terms of non-error response, we've already had one round of rethink on this in gh, where we went from returning the empty string to a list of length zero. And that ended up being a breaking change for usethis (but a positive change) Just want to make sure we don't make the same mistake again.

https://github.com/r-lib/gh/commit/6b735b192f44b7f7253a3a6cf7827ccab1060d54

Reply:
Another reason why I pulled this change into this PR is to make sure that the response content is returned _as is_ in the condition. `gh_error()` expects the response to be a list with a `message` item which it uses to construct the output message. After this PR, you won't need to look at the message to reason about the response, instead both the headers and the content are returned unprocessed in the condition object -- this is the intent behind calling `content(response)` again in the condition construction at the end of `gh_error()`.

I was trying to make minimal changes, but I'd be happy to refactor if you prefer another approach.

Reply:
I did not take in the whole picture, so definitely take my comment with a large grain of salt.